AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/test/Mutations/TestMutationHappyPath.dfy

```diff
@@ -0,0 +1,217 @@
+    var testInput := Types.ApplyMutationInput(
+      MutationToken := initializeToken,
+      PageSize := Some(24),
+      Strategy := Some(strategy),
+      SystemKey := Types.SystemKey.trustStorage(trustStorage := Types.TrustStorage()));
+    var applyOutput :- expect keyStoreAdminUnderTest.ApplyMutation(testInput);
+
+    expect applyOutput.MutationResult.CompleteMutation?, "Apply Mutation output should not continue!";
+
+    var expectedDecryptOnlyItems := if doNotVersion then
+      versionCount + 1
+    else
+      versionCount + 2;
+    verifyMutationResults(
+      storage := storage,
+      initialHV := initialHV,
+      keyStoreTerminal := keyStoreTerminal,
+      branchKeyIdentifier := branchKeyIdentifier,
+      mutationsRequest := mutationsRequest,
+      expectedDecryptOnlyItems := expectedDecryptOnlyItems
+    );
+
+    var _ := CleanupItems.DeleteBranchKey(Identifier:=branchKeyIdentifier, ddbClient:=ddbClient);
+  }
+
+  method verifyMutationResults(
+    storage: KeyStoreTypes.IKeyStorageInterface,
+    initialHV: KeyStoreTypes.HierarchyVersion,
+    keyStoreTerminal: KeyStoreTypes.IKeyStoreClient,
+    branchKeyIdentifier: string,
+    mutationsRequest: Types.Mutations,
+    expectedDecryptOnlyItems: int
+  )
+    requires storage.ValidState()
+    requires keyStoreTerminal.ValidState()
+    modifies storage.Modifies
+    modifies keyStoreTerminal.Modifies
+  {
+    var versionQuery := KeyStoreTypes.QueryForVersionsInput(
+      Identifier := branchKeyIdentifier,
+      PageSize := 24
+    );
+    var queryOut :- expect storage.QueryForVersions(versionQuery);
+    var items := queryOut.Items;
+    var (expectedKmsArn, expectedHV, expectedEncryptionContext) := getExpectedTerminalValues(mutationsRequest, initialHV);
+    expect
+      |items| == expectedDecryptOnlyItems,
+      "Test expects there to be " + String.Base10Int2String(expectedDecryptOnlyItems) + " Decrypt Only items! Found: " + String.Base10Int2String(|items|);
+
+    var itemIndex := 0;
+
+    var branchKeyVersionInputForStorage: KeyStoreTypes.GetEncryptedBranchKeyVersionInput;
+    var branchKeyVersionOutputFromStorage: KeyStoreTypes.GetEncryptedBranchKeyVersionOutput;
+    var branchKeyVersionInputForKeyStore: KeyStoreTypes.GetBranchKeyVersionInput;
+    var branchKeyVersionOutputFromKeyStore: KeyStoreTypes.GetBranchKeyVersionOutput;
+    var branchKeyVersionOutputEC: KeyStoreTypes.EncryptionContextString;
+
+    while itemIndex < |items|
+    {
+      var item := items[itemIndex];
+      expect
+        item.Type.HierarchicalSymmetricVersion?,
+        "Query for Decrypt Only returned a non-Decrypt Only!";
+      var versionUUID := item.Type.HierarchicalSymmetricVersion.Version;
+      expect "type" in item.EncryptionContext, "Decrypt Only item is missing 'type' from EC!!";
+      verifyTerminalProperties(item.EncryptionContext, expectedEncryptionContext, expectedKmsArn);
+
+      // Get branchKeyVersion from storage
```

Comment:
Keystore returns EC send to KMS (customer send EC without transformation) but storage returns a "actual" EC (with transformation)